libavfilter/dnn/dnn_backend_openvino.c

```diff
@@ -846,23 +806,46 @@ DNNReturnType ff_dnn_execute_model_async_ov(const DNNModel *model, DNNExecBasePa
         return DNN_ERROR;
     }
 
-    while (ff_queue_size(ov_model->inference_queue) >= ctx->options.batch_size) {
+    if (ctx->options.async) {
+        while (ff_queue_size(ov_model->inference_queue) >= ctx->options.batch_size) {
+            request = ff_safe_queue_pop_front(ov_model->request_queue);
+            if (!request) {
+                av_log(ctx, AV_LOG_ERROR, "unable to get infer request.\n");
+                return DNN_ERROR;
+            }
+
+            ret = execute_model_ov(request, ov_model->inference_queue);
+            if (ret != DNN_SUCCESS) {
+                return ret;
+            }
+        }
+
+        return DNN_SUCCESS;
+    }
+    else {
+        if (model->func_type == DFT_ANALYTICS_CLASSIFY) {
+            // Once we add async support for tensorflow backend and native backend,
+            // we'll combine the two sync/async functions in dnn_interface.h to
+            // simplify the code in filter, and async will be an option within backends.
+            // so, do not support now, and classify filter will not call this function.
+            return DNN_ERROR;
+        }
+
+        if (ctx->options.batch_size > 1) {
+            avpriv_report_missing_feature(ctx, "batch mode for sync execution");
```

Comment:
looks that batch mode for sync execution can be supported without much effort, right?  I do not mean you'd add such support or not, just a discussion.

Reply:
Yes. It seems like we can use the same code we do for async batch mode to support this. Just that we'd need to be sure that the flush frame function is being called for edge cases.

Reply:
i see

Reply:
I tried it also and it works fine since all filters which can use `batch_mode` call the flush frame function.

Reply:
we may enable it in another new patch some time later

Reply:
Sure